ts/utils/csv_export_utils.ts

```diff
@@ -0,0 +1,48 @@
+function convertToCSV(
+    headers: string[],
+    objArray: Array<{[key: string]: any}>): string {
+    let str = `${headers.join('\t')}\r\n`;
+
+    for (const element of objArray) {
+        let line = '';
+        for (const field of Object.keys(element)) {
+            if (line !== '') { line += '\t'; }
+            if (typeof(element[field] === 'object')) {
+                line += JSON.stringify(element[field]);
+            } else {
+                line += element[field];
+            }
+        }
+
+        str += `${line}\r\n`;
+    }
+
+    return str;
+}
+
+export function exportCSVFile(
```

Comment:
Super tiny nit, feel free to ignore: Fn naming is a bit ambiguous for return type, prefer more explicit fn names

`export function exportDataToCSVAndDownloadForUser(` or something (?)

Reply:
good nit!